salt/cloud/cli.py

```diff
@@ -402,13 +402,6 @@ def run(self):
                     self.config['pki_dir'], vm_['pub_key'], key_id
                 )
 
-            if 'os' not in vm_:
```

Comment:
I think you can take the above code block out here too, since you did move it into the other file.